### PR TITLE
[ML] Added datastore and job to allowed input types for Jobs

### PR DESCRIPTION
**Pull request by nthandeMS**
# Description
Resolves an issue where `azureml://datastores` and `azureml://job` inputs were not allowed as inputs to the `input` section of schemas. 

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by TonyJ1 — APPROVED** [review]
Do we need to update changelog?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/input_output_entry.py [hunk 1]
```diff
@@ -35,6 +35,8 @@ def generate_path_property(azureml_type):
     return UnionField(
         [
             ArmVersionedStr(azureml_type=azureml_type),
+            ArmVersionedStr(azureml_type=AzureMLResourceType.DATASTORE),
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/input_output_entry.py:38]: I believe these are the patterns we need to support. https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ml/azure-ai-ml/azure/ai/ml/constants/_common.py#:~:text=JOB_URI_FORMAT%20%3D%20%22azureml,azureml%3A//datastores/(%5B%5E/%5D%2B)/paths/(.%2B)%22

AzureML also has public docs on this, but it's not comprehensive unfortunately. https://learn.microsoft.com/en-us/azure/machine-learning/how-to-read-write-data-v2?tabs=cli#supported-paths

**Review by luigiw — CHANGES_REQUESTED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/input_output_entry.py (hunk 1 above)
Comment (luigiw) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/input_output_entry.py:38]: Can you also add some unit tests? Thx.

Comment (nthandeMS) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/input_output_entry.py]: Given this is an issue only raised by the VSCode extension, I believe we may be better off leaving testing to their end.

**Review by luigiw — APPROVED** [review]

**Pull request merged**